src/Grid/renderer.jsx

```diff
@@ -1,188 +1,102 @@
 import React, { Fragment } from 'react';
-import Switch from 'antd/lib/switch';
-import Popover from 'antd/lib/popover';
-import Input from 'antd/lib/input';
-import Form from 'antd/lib/form';
-import { Editor, CodeMirror, ListField, Upload } from '..';
-import { truncateText } from '../helper';
-
-export const getDisplay = (fieldType, record, dataIndex, children, maxLength) => {
-    if (fieldType === 'boolean') {
-        return <Switch disabled={true} checked={record}/>
-    }
+import styled from 'styled-components';
+import { Popover, Switch, Input, InputNumber } from 'antd';
+import { FormItem, Editor, CodeMirror, ListField, Upload, prettifyJson, truncateText } from '..';
+
+const Link = styled.span`
+    cursor: pointer;
+    text-decoration: underline;
+`;
+
+const CodeSnippet = ({ html, link, children }) => {
+    const content = (
+        <pre className="language-bash">
+            {html ? <div dangerouslySetInnerHTML={{ __html: children }}/> : children}
+        </pre>
+    );
+
+    return (
+        <Popover content={content}>
+            <Link>{link}</Link>
+        </Popover>
+    );
+};
 
-    if (fieldType === 'object') {
-        const content = (
-            <pre className="language-bash">
-                {JSON.stringify(record, null, 2)}
-            </pre>
-        );
+const ImagePreview = ({ data }) => {
+    let url, title;
+    let MoreLink = null;
 
-        return (
-            <Popover content={content} title={dataIndex}>
-                <span style={{ cursor: 'pointer', textDecoration: 'underline' }}>
-                    object
-                </span>
-            </Popover>
-        );
+    if (!data) {
+        return null;
     }
 
-    if (fieldType === 'html') {
-        const content = (
-            <pre className="language-bash">
-                <div dangerouslySetInnerHTML={{ __html: record }}/>
-            </pre>
-        );
-
-        return (
-            <Popover content={content} title={dataIndex}>
-                <span style={{ cursor: 'pointer', textDecoration: 'underline' }}>
-                    html
-                </span>
-            </Popover>);
+    if (Array.isArray(data) && data.length > 0) {
+        url = data[0].url;
+        title = data[0].name;
+        MoreLink = <span style={{ paddingLeft: '5px' }}>({data.length - 1} more)</span>;
```

Comment:
ein todo vielleicht dran oder ticket erstellen das wir das "more" in l10n aufnehmen

Reply:
done